### ISSUE TITLE
Obligation request returning error code 500

### ISSUE THREAD

**Issue by nigelrainer** [hmrc/vat-api#46]
We have been advised to contact you by SDS who seem unable to help.
Our code worked fine a few weeks ago and we could test obligation requests and submit test VAT returns.  This is the first time we've tried a test with the latest release of the VAT API.

Due to the clear out of the API by HMRC every month or so, we had to create a new user.
Our code gets the HMRC authentication token and then tries to make an obligations request.  The details at this stage are:

GET request = https://test-api.service.hmrc.gov.uk/vat/XXXXXXXXX/obligations?from=2017-10-01&to=2017-12-31&status=A

application/vnd.hmrc.1.0+json
application/json
Headers.Add("Authorization", "Bearer f85737e41b3c72f219b77282bae91");

(Where XXXXXXXXX is the VRN and the token received back from HMRC is added to the request headers.)

At this point the server returns the 500 error.  Can you help?

**Comment by RobertMarczycki** [comment]
It looks like 2018 date ranges work correctly.

**Comment by CoppensBart** [comment]
I have the same issue, I get the 500 internal server error, except when my date range is from 2018-01-01, to 2018-12-31.

**Comment by jonathanleather** [comment]
Hi
We have raised a story on the backlog to address the issue with the canned responses on the VAT obligations end point. See issue #62 for more information. 
It is not the years used that is the problem, it is the date range. 

For example:
from=2016-01-01 to=2016-07-01                  - Returns a successful response
from=2016-01-01 to=2016-06-30                  - Returns a 500

We currently do not have a date for when this will be implemented into ET.

@CoppensBart the date ranges you are using should work. I have just tested this in ET and received a 200. Can you double check your request and try again? Let me know if you have any further issues.

Thanks
Jon

**Comment by hairyfingers** [comment]
Hi.

@jonathanleather Can I just clarify - you are acknowledging there is an issue here, just you're not sure when it will be resolved? or that you will simply change the error code/message to say invalid date range?

If the former, then fine I can wait on the fix.  If the latter, are we therefore required to prevent the user from selecting a range of 6 months or less.

for the record I've got the same issue

for example
from=2016-01-01 to=2016-07-01 - Returns a successful response
from=2016-01-01 to=2016-06-30 - Returns a 500

from=2018-07-01 to=2019-01-01 - Returns a successful response
from=2018-07-01 to=2018-12-31 - Returns a 500

basically any 6 month period returns 500 but go 1 day further and it's fine.

thanks

**Comment by jonathanleather** [comment]
Hi @hairyfingers 

Yes there is an issue here. (all of the date ranges you stated should return a successful response)
There should not be a limitation on the minimum number of days but there is one for the maximum number of days (1 year).

Hope that clears it up for you.

Cheers
Jon
